ui/app/templates/components/select-dropdown.hbs

```diff
@@ -0,0 +1,24 @@
+<div class="field">
+  <label
+    for="select-dropdown-{{this.elementId}}"
+    class="is-label"
+    data-test-select-label
+    >
+    {{dropdownLabel}}
+  </label>
+  <div class="control {{if isInline "select is-inline-block"}}">
+    <div class="select {{if isFullwidth "is-fullwidth"}}">
+    <select
+      class="select"
+      id="select-dropdown-{{this.elementId}}"
+      name="selectedItem"
+      onchange={{action "onSelectItem"}}
+      data-test-select-dropdown
+      >
+      {{#each options as |op|}}
```

Comment:
I'm wondering if the component should be refactored to render in block format so the options are more customizable. Here, we assume that options is an array of values, but looking through our existing `<select>`s, most options are an array of objects where the `<select value="{{option.some_arbitrary_key}}>"`. Any thoughts on whether this component should be flexible or should require that options are formatted in the parent component?

Reply:
I think we still have options as an array, but optionally an array of objects. There are many places where we have `{value: 'foo', label: 'Foo'}` for options and yielding a block would work, but then you'd also have to yield the selected item and the consumer would only be able to render an `option` tag anyway. Given that - what do you think about having a `valueAttribute` or `optionValueName` (I'm not sure what exactly to call it) that defaults to `value` and then in the each here you could do `<option value={{get op valueAttribute}}>` and have a similar thing for the text in the option (if option is an object). ? I guess the question then will be: how do you trigger that behavior? Should the component just detect if it's an array of strings or an array of objects, or should there be another attribute to trigger it ?

Reply:
yes, this is a great suggestion since it'll make the component more flexible. i've updated it to handle options as either an array of strings or objects. thank you for the idea!